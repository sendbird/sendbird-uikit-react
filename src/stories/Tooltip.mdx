import { Meta, Canvas, Controls } from '@storybook/addon-docs';
import Tooltip from '../ui/Tooltip';
import TooltipWrapper from '../ui/TooltipWrapper';
import * as TooltipStories from './Tooltip.stories';

<Meta title="UI/Tooltip" component={Tooltip} of={TooltipStories} />

# Tooltip and TooltipWrapper

`Tooltip` and `TooltipWrapper` components provide text labels that help explain the function of a button or other user interface action.

<Canvas of={TooltipStories.Default} />
<Controls />

### Importing Tooltip and TooltipWrapper

To use these components in your application, import them as follows:

```tsx
import Tooltip from '@sendbird/uikit-react/ui/Tooltip';
import TooltipWrapper from '@sendbird/uikit-react/ui/TooltipWrapper';
```

### TooltipWrapper Examples
#### Hover Tooltip
A tooltip that appears when the user hovers over the element.
<Canvas of={TooltipStories.HoverTooltip} />
