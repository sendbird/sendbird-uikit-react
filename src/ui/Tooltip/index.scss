@import '../../styles/variables';

.sendbird-tooltip {
  position: relative;
  display: inline-flex;
  justify-content: center;
  max-width: 200px;
  border-radius: 8px;
  padding: 8px 16px;
  @include themed() {
    background-color: t(bg--7-3);
  }

  &::after {
    content: '';
    position: absolute;
    bottom: -4px;
    width: 8px;
    height: 8px;
    transform: rotate(45deg);
    @include themed() {
      background-color: t(bg--7-3);
    }
  }

  &__text {
    font-family: var(--sendbird-font-family-default);
    font-size: 12px;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: 1.33;
    letter-spacing: normal;
  }
}
