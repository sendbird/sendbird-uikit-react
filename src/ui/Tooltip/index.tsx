import React, { ReactElement } from 'react';

import './index.scss';

import Label, { LabelColors, LabelTypography } from '../Label';

export interface TooltipProps {
  className?: string | Array<string>;
  children?: string | ReactElement;
}
export default function Tooltip({
  className = '',
  children= '',
}: TooltipProps): ReactElement {
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-tooltip',
      ].join(' ')}
    >
      <Label
        className="sendbird-tooltip__text"
        type={LabelTypography.CAPTION_2}
        color={LabelColors.ONCONTENT_1}
      >
        {children}
      </Label>
    </div>
  );
}
