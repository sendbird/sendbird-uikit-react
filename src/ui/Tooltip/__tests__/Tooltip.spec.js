import React from 'react';
import { render } from '@testing-library/react';

import Tooltip from "../index";

describe('ui/Tooltip', () => {
  it('should have className', function () {
    const className = "example-text";
    const text = 'Sravan S, Hoon Baek, Chong Bu, Mickey, Leo Shin, Doo Rim, Tez Park, Harry Kim, and you';
    const { container } = render(
      <Tooltip className={className}>
        {text}
      </Tooltip>
    );
    expect(
      container.querySelectorAll('.sendbird-tooltip')
    ).toHaveLength(1);
  });

  it('should do a snapshot test of the ReactedUsers DOM', function () {
    const className = "example-text";
    const text = 'Sravan S, Hoon Baek, Chong Bu, Mickey, Leo Shin, Doo Rim, Tez Park, Harry Kim, and you';
    const { asFragment } = render(
      <Tooltip className={className}>
        {text}
      </Tooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
