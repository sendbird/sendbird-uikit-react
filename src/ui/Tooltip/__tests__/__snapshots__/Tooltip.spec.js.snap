// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Tooltip should do a snapshot test of the ReactedUsers DOM 1`] = `
<DocumentFragment>
  <div
    class="example-text sendbird-tooltip"
  >
    <span
      class="sendbird-tooltip__text sendbird-label sendbird-label--caption-2 sendbird-label--color-oncontent-1"
    >
      Sravan S, Hoon Baek, Chong Bu, Mickey, Leo Shin, Doo Rim, Tez Park, Harry Kim, and you
    </span>
  </div>
</DocumentFragment>
`;
